subworkflows/local/prepare_genome/main.nf

```diff
@@ -27,31 +27,32 @@ include { UNZIP as UNZIP_RT                      } from '../../../modules/nf-cor
 
 workflow PREPARE_GENOME {
     take:
-        ascat_alleles           // channel: [optional]  ascat allele files
-        ascat_loci              // channel: [optional]  ascat loci files
-        ascat_loci_gc           // channel: [optional]  ascat gc content file
-        ascat_loci_rt           // channel: [optional]  ascat replictiming file
-        chr_dir                 // channel: [optional]  chromosome files
-        dbsnp                   // channel: [optional]  dbsnp
-        fasta                   // channel: [mandatory] fasta
-        fasta_fai               // channel: [optional]  fasta_fai
-        germline_resource       // channel: [optional]  germline_resource
-        known_indels            // channel: [optional]  known_indels
-        known_snps              // channel: [optional]  known_snps
-        pon                     // channel: [optional]  pon
+    ascat_alleles     // channel: [optional]  ascat allele files
+    ascat_loci        // channel: [optional]  ascat loci files
+    ascat_loci_gc     // channel: [optional]  ascat gc content file
+    ascat_loci_rt     // channel: [optional]  ascat replictiming file
+    chr_dir           // channel: [optional]  chromosome files
+    dbsnp             // channel: [optional]  dbsnp
+    fasta             // channel: [mandatory] fasta
+    fasta_fai         // channel: [optional]  fasta_fai
+    germline_resource // channel: [optional]  germline_resource
+    known_indels      // channel: [optional]  known_indels
+    known_snps        // channel: [optional]  known_snps
+    pon               // channel: [optional]  pon
 
 
     main:
+    fasta = fasta.map{ fasta -> [ [ id:fasta.baseName ], fasta ] }
+    versions = Channel.empty()
 
-    ch_versions = Channel.empty()
+    BWAMEM1_INDEX(fasta)     // If aligner is bwa-mem
+    BWAMEM2_INDEX(fasta)     // If aligner is bwa-mem2
+    DRAGMAP_HASHTABLE(fasta) // If aligner is dragmap
 
```

Comment:
Still no love for `snap-aligner` 🥺 ?

Reply:
We need more time for that, but it'll come.
Do we have a github issue for that?

Reply:
Don't think so, not all that important too ;)

Reply:
sorry no, more important at the moment to make sure the existing aligners all play nice with the variantcallers we have. If you open an issue we can pencil it in eventually